locksmith/__tests__/controllers/v2/userController.test.ts

```diff
@@ -0,0 +1,48 @@
+import { vi, expect } from 'vitest'
+import request from 'supertest'
+import app from '../../app'
+import { IssueUserTokenOptions } from '@coinbase/waas-server-auth'
+
+const googleAuthToken = 'token'
+const coinbaseAuthToken = 'coinbase'
+const emailAddress = 'test@test.com'
+const selectedProvider = 'GOOGLE_ACCOUNT'
+
+vi.mock('../../../src/utils/verifyGoogleToken', () => {
+  return {
+    verifyToken: vi.fn().mockImplementation((email: string, token: string) => {
+      return googleAuthToken === token
+    }),
+  }
+})
+
+vi.mock('@coinbase/waas-server-auth', () => {
+  return {
+    issueUserToken: vi
+      .fn()
+      .mockImplementation((options: IssueUserTokenOptions): Promise<string> => {
+        return Promise.resolve(coinbaseAuthToken)
+      }),
+  }
+})
+
+describe('Get UUID from Coinbase WAAS', () => {
+  it('returns UUID from Coinbase Waas', async () => {
+    expect.assertions(2)
+    const retrieveWaasUuidRes = await request(app)
+      .post(`/users/${emailAddress}/${selectedProvider}/waas`)
+      .send({ token: googleAuthToken })
+
+    expect(retrieveWaasUuidRes.status).toBe(200)
+    expect(retrieveWaasUuidRes.body).toEqual({ token: coinbaseAuthToken })
+  })
+
+  it('returns error if googleAuthToken is invalid', async () => {
```

Comment:
Can you add a test for when _no_ token is provided?

Reply:
Also a test when the emai address does not correspond to a valid account on our end?

Reply:
and a test when it is an email for a legacy unlock account?